source/rust_verify/src/rust_to_vir_func.rs

```diff
@@ -162,23 +172,186 @@ pub(crate) fn check_item_fn<'tcx>(
     generics: &'tcx Generics,
     body_id: CheckItemFnEither<&BodyId, &[Ident]>,
 ) -> Result<Option<Fun>, VirErr> {
-    let path = def_id_to_vir_path(ctxt.tcx, id);
-    let name = Arc::new(FunX { path: path.clone(), trait_path: trait_path.clone() });
+    let this_path = def_id_to_vir_path(ctxt.tcx, id);
+
+    let is_verus_spec = this_path.segments.last().expect("segment.last").starts_with(VERUS_SPEC);
+    let is_new_strlit =
+        ctxt.tcx.is_diagnostic_item(Symbol::intern("pervasive::string::new_strlit"), id);
 
     let vattrs = get_verifier_attrs(attrs)?;
+    let mode = get_mode(Mode::Exec, attrs);
+
+    let (path, proxy, visibility) = if vattrs.external_fn_specification {
```

Comment:
Would it make sense to factor out this branch of the `if`, to clarify that it handles proxies and to avoid making this function even longer?

Reply:
sure